### PR TITLE
chore: upgrade drf

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

At some point we're going to want to upgrade to Django 4

Upgrading DRF is a prerequisite for that, and probably good in and of itself

## Changes

* bumps DRF
* bumps a few dev dependencies

## How did you test this code?

```
pip install --upgrade pip
pip install --upgrade pip-tools
pip-compile --rebuild requirements.in && pip-compile --rebuild requirements-dev.in
pip install -r requirements.txt -r requirements-dev.txt  
```

clicking around locally and seeing things seem to work

then opening this PR to see whether tests fail

### CODE REVIEW

**Review by Twixes — APPROVED** [review]
Looks good based on https://www.django-rest-framework.org/community/release-notes/#313x-series!

**Review by benjackwhite — APPROVED** [review]
No complaints about the code - hopefully anything wrong is caught by tests 😬

**Review by benjackwhite — APPROVED** [review]
No complaints about the code - hopefully anything wrong is caught by tests 😬

File: .github/workflows/ci-backend.yml [hunk 1]
```diff
@@ -132,8 +131,8 @@ jobs:
 
             - name: Check static typing
               run: |
-                  cd current
-                  mypy .
+                  cd current
```
Comment (pauldambra) [inline .github/workflows/ci-backend.yml:134]:
```suggestion
                  cd current
```

**Pull request merged**